csharp/src/Apache.Arrow/Ipc/ArrowFileReaderImplementation.cs

```diff
@@ -70,36 +72,79 @@ protected override async Task ReadSchemaAsync()
             {
                 BaseStream.Position = BaseStream.Length - ArrowFileConstants.Magic.Length - 4;
 
-                bytesRead = await BaseStream.ReadAsync(buffer, 0, 4).ConfigureAwait(false);
-                footerLength = BinaryPrimitives.ReadInt32LittleEndian(buffer);
+                bytesRead = await BaseStream.ReadAsync(buffer.AsMemory(0, 4)).ConfigureAwait(false);
```

Comment:
You'd be better off moving the declaration of bytesRead into each of these delegates so they don't have to close over the shared field and instead it can just be a local (that doesn't need to be lifted to any state machine).

Reply:
Good call. Fixed.